### PR TITLE
Merge 3.6 to main

### PR DESCRIPTION
**Pull request by Aflynn50**
Merge up 3.6 to main. The commits in the are:
- #18241 from hpidcock
  - #18219 from hpidcock 
    - Conflicts in apiserver and debuglog, resovled to keep new closing channel functionality
  - #18199 from manadart
  - #18230 from aflynn50
  - Plus other forward merges.
- #18227 from Aflynn50
  - #18170 from Aflynn50
    - Conflict in tests/main.sh, resolved to add both new flags (B and V)
  - #18212 from gfouillet
    - Had conflicts in run.go, removed code that was not relevant anymore (needs checking @gfouillet)
- #18216 from Aflynn50
- #18211 from Aflynn50
- #18177 from hpidcock
- #18176 from cuishuang
- #18196 from juju/dependabot/github_actions/actions
- #18178 from hpidcock

### CODE REVIEW

File: internal/worker/dbaccessor/tracker.go [hunk 1]
```diff
@@ -380,10 +389,10 @@ func (w *trackedDBWorker) loop() error {
 // ensureDBAliveAndOpenNewIfRequired is a bit long-winded, but it is a way to
 // ensure that the underlying database is alive and well. If it is not, we
 // attempt to open a new one. If that fails, we return an error.
-func (w *trackedDBWorker) ensureDBAliveAndOpenIfRequired(db *sql.DB) (*sql.DB, error) {
+func (w *trackedDBWorker) ensureDBAliveAndOpenIfRequired(ctx context.Context, db *sql.DB) (*sql.DB, error) {
 	// Allow killing the tomb to cancel the context,
 	// so shutdown/restart can not be blocked by this call.
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
 	ctx = w.tomb.Context(ctx)
```
Comment (gfouillet) [inline internal/worker/dbaccessor/tracker.go:396]: **todo:** This line should be removed as well (bounding context to tomb is now handled outside the function l343:

https://github.com/juju/juju/pull/18231/files#diff-4dcf535ef6cc5d3f34ed25bc3d44275803733410d9e05274130ac53d89ce206fR343

**Review by gfouillet — CHANGES_REQUESTED** [review]
On small change please.

File: internal/worker/dbaccessor/tracker.go (hunk 1 above)
Comment (Aflynn50) [reply internal/worker/dbaccessor/tracker.go]: Cool, deleted that wrapper

**Review by nvinuesa — APPROVED** [review]
LGTM

**Review by hpidcock — APPROVED** [review]
My stuff looks good. Thanks for doing this.

**Pull request merged**